src/lua/thread.rs

```diff
@@ -155,30 +154,33 @@ fn main_loop(receiver: Receiver<LuaMessage>, lua: &mut Lua) {
             }
             Ok(message) => {
                 trace!("Handling a request");
-                handle_message(message, lua);
+                if !handle_message(message, lua) {
+                    return
+                }
             }
         }
     }
 }
 
 /// Handle each LuaQuery option sent to the thread
-fn handle_message(request: LuaMessage, lua: &mut Lua) {
+fn handle_message(request: LuaMessage, lua: &mut Lua) -> bool {
```

Comment:
Should this maybe be a `Result`? The bool here makes it a little implicit that `false` here means we are supposed to restart, and isn`t an error from handling the message